### PR TITLE
Bugs/120

### PR DESCRIPTION
**Pull request by frantsao**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions
* Remember to set **idealista:develop** as base branch;

### Description of the Change

*  Installs the latest package available for the Java major release by default
* Alternatives are managed by post-install package actions
* Upgraded to molecule 3 and goss 0.3.11
* Added support for Debian Buster, Ubuntu 20.04 and Centos 8.

### Benefits

* Improve reliability using the role in deployments and tests
* Security fixes

### Possible Drawbacks
N/A

### Applicable Issues
#120 #107 #114

### CODE REVIEW

**Comment by dortegau** [comment]
It would be nice to update `meta/main.yml` file

File: README.md [hunk 1]
```diff
@@ -22,18 +22,18 @@ This Ansible Role installs java ([OpenJDK](http://openjdk.java.net/) in a [Debia
...
 ### Prerequisities
 
 To use this role as dependency in your playbook, prerequisites below:
 
-Ansible 2.8.4.0 version installed.
+Ansible 2.9.x.x version installed.
 Inventory destination should be a Debian/Ubuntu or CentOS environment.
 
-For testing purposes you will need [Python 3.6+](https://www.python.org/downloads/release/python-368/) and [Pipenv](https://github.com/pypa/pipenv)
+For testing purposes you will need [Python 3.6+](https://www.python.org/downloads/release/python-368/), [Pipenv](https://github.com/pypa/pipenv) and [Molecule](https://github.com/ansible-community/molecule)
```
Comment (dortegau) [inline README.md:36]: AFAIK as molecule is defined in requirements file, you don't have to have molecule installed

File: README.md [hunk 2]
```diff
@@ -42,19 +42,19 @@ Create or add to your roles dependency file (e.g requirements.yml):
 ```yml
 - src: http://github.com/idealista/java_role.git
   scm: git
-  version: 5.0.0
+  version: 5.1.0
```
Comment (dortegau) [inline README.md:45]: I think that this version should be update when we expect to release a new version.

File: README.md [hunk 3]
```diff
@@ -105,28 +105,32 @@ A specific OpenJDK version should be selected overriding `java_open_jdk_version`
 Operative System | OpenJDK version
 --- | ---
 Debian Jessie | `8u171-b11-1~bpo8+1` (default)
```
Comment (dortegau) [inline README.md:107]: As version package is not defined, I think that this Readme section should be removed

File: README.md (hunk 1 above)
Comment (dortegau) [inline README.md:36]: As we are testing with Python 3.7, I prefer to tell that the user will need Python 3.7+

File: README.md (hunk 3 above)
Comment (frantsao) [reply README.md]: I changed the table in order to show the OpenJDK major releases supported.

**Pull request merged**